aten/src/ATen/Dispatch.h

```diff
@@ -2,17 +2,59 @@
 
 #include <ATen/core/DeprecatedTypeProperties.h>
 #include <ATen/core/Tensor.h>
+#include <ATen/record_function.h>
 #include <c10/macros/Macros.h>
 #include <c10/util/Exception.h>
 #include <c10/util/Half.h>
+#include <c10/util/Metaprogramming.h>
 #include <c10/util/complex.h>
+#include <c10/util/string_view.h>
+
+namespace at {
+/**
+ * The method should_include_kernel_dtype() returns true/false
+ * based on whether the switching code for a specific dtype should be
+ * included based on build time constants generated from tracing model
+ * execution. This method will be implmeneted via code-generation and
+ * included in this file when code-gen is ready.
+ */
```

Comment:
Hey @dhruvbird, is this comment still relevant? I see in master now we still have the comment, but we also have a mobile ifdef that adds the codegen'd version of the function:

```
#ifdef TEMPLATE_SELECTIVE_BUILD
#include <ATen/selected_mobile_ops.h>
#else
namespace at {
/**
 * The method should_include_kernel_dtype() returns true/false
 * based on whether the switching code for a specific dtype should be
 * included based on build time constants generated from tracing model
 * execution. This method will be implmeneted via code-generation and
 * included in this file when code-gen is ready.
 */
inline constexpr bool should_include_kernel_dtype(
  const char *kernel_tag_str,
  at::ScalarType scalar_type
) {
  return true;
}
}
#endif
```

So is `should_include_kernel_dtype()` always going to be a no-op for server builds? (cc @guoyejun)

Reply:
Yes, the comment is slightly out of date. Basically this code:

```
inline constexpr bool should_include_kernel_dtype(
  const char *kernel_tag_str,
  at::ScalarType scalar_type
) {
  return true;
}
```

will be used for server builds, and the code from the included file `<ATen/selected_mobile_ops.h>` will be used for mobile builds.

> So is should_include_kernel_dtype() always going to be a no-op for server builds? (cc @guoyejun)

Yes - basically any build that doesn't define the macro `TEMPLATE_SELECTIVE_BUILD`.

Reply:
got it thanks :)

Reply:
thanks